### DISCUSSION TITLE
Autogen Studio with Ollama Error.  Autogen still wanting access to OpenAI

### DISCUSSION THREAD

**Discussion by MrMWith** [microsoft/autogen#1931] [Q&A]
I have Ollama serving LLMs to Autogen Studio via the Model setup.  They test perfectly and show connected.  Each of the Agents within in the Workflow are exclusively using Agents with Ollama models.  There is no reference to OpenAI anywhere within the GroupChat Workflow.

The Workflow begins the task and the Agents are responding via the Ollama connection however, I am I getting an OpenAI error when it shouldn't be called at all? See below.

(yes, the RateLimitError is what is driving me to use local Ollama models.  I will go broke testing Autogen if I keep using the OpenAI API)

openai.RateLimitError: Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}

Note: I am on Windows running Autogen Studio in a conda env.

**Comment by descention** [comment]
Have you updated the workflow to use the updated agent? I've experienced an existing workflow will not have configuration changes propagated from agent/models.

**Reply by MrMWith** [reply]
Thanks for the reply. Yes, I have updated the agent and workflow to use Ollama models... as well as the primary assistant.  I wouldn't know this if not for already maxing out my API quota... strange.

**Reply by MrMWith** [reply]
I think I found the problem.  The Primary Assistant had 2 skills associated with it.  I removed the skills and it appears to be working now without accessing OpenAI.

**Reply by mdkberry** [reply]
Same for me I am getting successful model test with Ollama using mistral but when I try to run the playground it gives _ValueError: api_key is not present in llm_config or OPENAI_API_KEY env variable for agent ** adcopy_critic_agent**. Update your workflow to provide an api_key to use the LLM._

I tried deleting the skills and removing them from the primary assistant but still getting the message so didnt fix in my case. adcopy_critic_agent uses no skills and is assigned to the local LLM. I tried restart everything. still no luck.

**Reply by Pgz1975** [reply]
Had the same issue running Autogen studio from conda.
I deleted OPENAI_API_KEY in powershell with
Remove-Item Env:OPENAI_API_KEY
went back to conda and run autogen studio : all good

**Reply by viniljose** [reply]
I tried the below after the save of the default agent in settings didnt get reflected for Ollama even though the test was successful.
Ensure OPENAI_API_KEY is not the env
Went to Gallery and added Ollama model with below JSON payload,then tried using that model in Team Builder than open AI one
```
{
  "provider": "autogen_ext.models.ollama.OllamaChatCompletionClient",
  "component_type": "model",
  "version": 1,
  "component_version": 1,
  "description": "Chat completion client for Ollama hosted models.",
  "label": "OllamaChatCompletionClient",
  "config": {
    "model": "llama3.2:latest",
    "api_key": "ollama",
    "model_info": {
      "vision": false,
      "function_calling": true,
      "json_output": false,
      "family": "unknown",
      "structured_output": true
    },
    "base_url": "http://localhost:11434/v1"
  }
}
```
